docs/tools/prism-eslint-hook.js

```diff
@@ -0,0 +1,179 @@
+"use strict";
+
+const Prism = require("prismjs");
+const { Linter } = require("../../lib/api");
+const astUtils = require("../../lib/shared/ast-utils");
+
+/** @typedef {import("../../lib/shared/types").ParserOptions} ParserOptions */
+
+/**
+ * A set of content that needs to be marked.
+ * @type {Map<string, ParserOptions>}
+ */
+const contentsMustBeMarked = new Map();
+
+/**
+ * Add content that needs to be marked.
+ * @param {string} content Source code content that marks eslint errors.
+ * @param {ParserOptions} options The options used for validation.
+ * @returns {void}
+ */
+function addContentMustBeMarked(content, options) {
+    contentsMustBeMarked.set(content, options);
+}
+
+/**
+ * Register a hook for `Prism` to mark errors in ESLint.
+ * @returns {void}
+ */
+function installPrismESLintMarkerHook() {
+    const linter = new Linter({ configType: "flat" });
+
+    Prism.hooks.add("after-tokenize", env => {
+
+        const parserOptions = contentsMustBeMarked.get(env.code);
+
+        if (!parserOptions) {
+
+            // Ignore unmarked content.
+            return;
+        }
+        contentsMustBeMarked.delete(env.code);
+
+        const code = env.code;
+        const tokens = env.tokens;
+
+        /** Copied from SourceCode constructor */
+        const lineStartIndices = [0];
+        const lineEndingPattern = astUtils.createGlobalLinebreakMatcher();
+        let match;
+
+        while ((match = lineEndingPattern.exec(code))) {
+            lineStartIndices.push(match.index + match[0].length);
+        }
+
+        /**
+         * Converts a (line, column) pair into a range index.
+         * @param {Object} loc A line/column location
+         * @param {number} loc.line The line number of the location (1-indexed)
+         * @param {number} loc.column The column number of the location (1-indexed)
+         * @returns {number} The range index of the location in the file.
+         * Copied from SourceCode#getIndexFromLoc
+         */
+        function getIndexFromLoc(loc) {
+            const lineStartIndex = lineStartIndices[loc.line - 1];
+            const positionIndex = lineStartIndex + loc.column - 1;
+
+            return positionIndex;
+        }
+
+        /*
+         * Run lint to extract the error range.
+         */
+        const messages = linter.verify(
+
+            // Remove trailing newline and presentational `⏎` characters
+            code.replace(/⏎(?=\n)/gu, "").replace(/\n$/u, ""),
```

Comment:
Shall we merge this expression with [this one](https://github.com/eslint/eslint/blob/5a8efd5b7ad13eb320a1f468d1d4ab3c8ab99214/docs/tools/markdown-it-rule-example.js#L75-L78) into a separate function, to ensure that they don't get out of sync if we make a change?

Alternatively, we could extend `addContentMustBeMarked` to accept `code` as a new argument [here](https://github.com/eslint/eslint/blob/94a090719015448b7ede6204225734020f02b152/docs/.eleventy.js#L198-L200), so we could store it to use it later.